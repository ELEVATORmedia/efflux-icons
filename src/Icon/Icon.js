import React from 'react';
import PropTypes from 'prop-types';
import { buildPaths } from '../utils/Parser';

const Icon = (props) => {
    let { name, size, color } = props;

    let paths = buildPaths(name, color);

    if(paths.length < 1){
        return <p>?</p>;
    }

    return (
        <svg width={size} height={size} viewBox="0 0 1024 1024">
            {paths}
        </svg>
    );
};

Icon.propTypes = {
    name: PropTypes.string.isRequired,
    size: PropTypes.number,
    color: PropTypes.string,
    fill: PropTypes.bool,
};

Icon.defaultProps = {
    size: 20,
    color: 'black',
};

export default Icon;
