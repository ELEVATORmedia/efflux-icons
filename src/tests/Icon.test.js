import React from 'react';
import renderer from 'react-test-renderer';
import Icon from 'index';

describe('Icon component', () => {
    it('renders default without crashing', () => {
        const defaultIcon = renderer.create(<Icon name={'Home'} />);
        let tree = defaultIcon.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders blue icon crashing', () => {
        const defaultIcon = renderer.create(<Icon name={'Home'} color={'Blue'} />);
        let tree = defaultIcon.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders blue icon at 25px without crashing', () => {
        const defaultIcon = renderer.create(
            <Icon name={'Home'} color={'Blue'} size={25} />
        );
        let tree = defaultIcon.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
