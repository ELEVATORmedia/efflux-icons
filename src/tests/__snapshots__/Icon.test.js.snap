// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders blue icon at 25px without crashing 1`] = `
<svg
  height={25}
  viewBox="0 0 1024 1024"
  width={25}
>
  <path
    d="M669.355 998.4l252.245-0c42.415 0 76.8-34.385 76.8-76.8v-505.352c0-24.886-12.058-48.229-32.352-62.631l-454.048-322.225-454.047 322.225c-20.294 14.402-32.352 37.745-32.352 62.631l-0.001 505.352c0 42.416 34.385 76.8 76.8 76.8h230.4v-241.322c0-92.937 75.34-168.278 168.278-168.278s168.278 75.34 168.278 168.278v241.322z"
    style={
      Object {
        "fill": "none",
        "stroke": "Blue",
        "strokeLinecap": "butt",
        "strokeLinejoin": "miter",
        "strokeMiterlimit": "4",
        "strokeWidth": 51.2,
      }
    }
  />
</svg>
`;

exports[`Icon component renders blue icon crashing 1`] = `
<svg
  height={20}
  viewBox="0 0 1024 1024"
  width={20}
>
  <path
    d="M669.355 998.4l252.245-0c42.415 0 76.8-34.385 76.8-76.8v-505.352c0-24.886-12.058-48.229-32.352-62.631l-454.048-322.225-454.047 322.225c-20.294 14.402-32.352 37.745-32.352 62.631l-0.001 505.352c0 42.416 34.385 76.8 76.8 76.8h230.4v-241.322c0-92.937 75.34-168.278 168.278-168.278s168.278 75.34 168.278 168.278v241.322z"
    style={
      Object {
        "fill": "none",
        "stroke": "Blue",
        "strokeLinecap": "butt",
        "strokeLinejoin": "miter",
        "strokeMiterlimit": "4",
        "strokeWidth": 51.2,
      }
    }
  />
</svg>
`;

exports[`Icon component renders default without crashing 1`] = `
<svg
  height={20}
  viewBox="0 0 1024 1024"
  width={20}
>
  <path
    d="M669.355 998.4l252.245-0c42.415 0 76.8-34.385 76.8-76.8v-505.352c0-24.886-12.058-48.229-32.352-62.631l-454.048-322.225-454.047 322.225c-20.294 14.402-32.352 37.745-32.352 62.631l-0.001 505.352c0 42.416 34.385 76.8 76.8 76.8h230.4v-241.322c0-92.937 75.34-168.278 168.278-168.278s168.278 75.34 168.278 168.278v241.322z"
    style={
      Object {
        "fill": "none",
        "stroke": "black",
        "strokeLinecap": "butt",
        "strokeLinejoin": "miter",
        "strokeMiterlimit": "4",
        "strokeWidth": 51.2,
      }
    }
  />
</svg>
`;
